internal/internal_task_handlers.go

```diff
@@ -796,9 +800,27 @@ processWorkflowLoop:
 	}
 	errRet = err
 	completeRequest = response
+	resetter = workflowContext.SetPreviousStartedEventID
 	return
 }
 
+// indexInvocations builds a map of the interaction invocations contained in a
+// workflow task where the index is the interaction's event level (i.e. the
+// event after which the interaction can execute) and the value is the
+// invocation itself. This function may return an empty map but it will not
+// return nil.
+func indexInvocations(workflowTask *workflowTask) map[int64][]*interactionpb.Invocation {
+	out := map[int64][]*interactionpb.Invocation{}
+	for _, ev := range workflowTask.task.GetHistory().GetEvents() {
```

Comment:
Can you help me understand this a bit? Can you confirm that `GetEvents()` is _only_ events for this task?

Right now what I read is "every time you see a workflow task started, duplicate the interaction list on its meta event ID in the map" which seems a bit strange. Is the goal to see if the _most recent_ workflow task started matches the meta event ID (missing conditional?) and then only add that to the map?

As the algorithm sits now, why even loop over history events? Just make the map from the interactions.

Reply:
Thank you for catching this. This code block is from when we were going to attach invocations to the WFTStartedEvent. We have switched to attaching invocations directly to the WFT. I included the wrong impl in this PR. Updated this function and added a UT.